### PR TITLE
Add metric metadata endpoints

### PR DESCRIPTION
**Pull request by flapjack103**
- Adds GET and PUT operations for the metadata endpoint.
- Some confusion around https://github.com/DataDog/datadogpy/compare/agrant/metrics_metadata?expand=1#diff-0741b06f081c976a404268a44c97cc2fR15 and if this should be `metrics` or `metadata` ?

### CODE REVIEW

File: datadog/api/metadata.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        :type statsd_interval: integer
+
+        :return: JSON response from HTTP request
+
+        >>> api.Metadata.update(metric_name='api.requests.served', metric_type="counter")
+        """
+        if not metric_name:
+            raise KeyError("'metric_name' parameter is required")
+
+        return super(Metadata, cls)._trigger_class_action('PUT', 'metrics', metric_name, **params)
```
Comment (yannmh) [inline datadog/api/metadata.py]: Can we simply do 
```
return super(Metadata, cls).update(id=metric_name, **params)
```
?

File: datadog/api/metadata.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# datadog
+from datadog.api.resources import GetableAPIResource, CreateableAPIResource, \
+    UpdatableAPIResource, ActionAPIResource
+
+
+class Metadata(GetableAPIResource, CreateableAPIResource, UpdatableAPIResource, ActionAPIResource):
```
Comment (flapjack103) [inline datadog/api/metadata.py]: I added some clarification in the update docstring. The update request will create the metadata if it doesn't exist.

**Comment by flapjack103** [comment]
Thanks for the review and fixing the violations. Let me know if the changes are good or if it needs anything else.

**Comment by yannmh** [comment]
Looks great, thanks @flapjack103 🙇

**Pull request merged**